system/inc/active_object.h

```diff
@@ -280,6 +280,10 @@ class ActiveObjectBase
         return _thread_id == std::this_thread::get_id();
     }
 
+    void setCurrentThread() {
```

Comment:
could you make this a protected method - seems to make sense from the subclass rather than general public access.